base/resources/scripts/languagecheck.js

```diff
@@ -0,0 +1,40 @@
+/* Nodejs to set up a server on 8081 to modify the argv.json file
+   used by in-browser VSCode. Uses the accept-language header.
+   Also used by jupyter to determine language.
+ */
+var http = require("http");
+var fs = require("fs");
+
+http
+  .createServer(function (req, res) {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    var lang = req.headers["accept-language"];
+    lang ? lang : (lang = "en");
+    lang = lang.substring(0, 2);
+
+    lang === "fr" || lang === "en" ? lang : (lang = "en");
+
+    // READ THE FILE AND COMPARE TO LANG IF IT CHANGES DO THE FOLLOWING, SYNC to not write to it as well
+    var file = fs.readFileSync(
+      "/home/jovyan/.config/Code/User/argv.json",
+      "utf-8"
+    ); // must specify encoding to return string
+    var browserLang = '"locale":"' + lang + '"'; //search using this, "locale":"en" if browserLang not in file then change
+
+    if (file.indexOf(browserLang) == -1) {
+      // if the file index of browserLang == -1 (meaning not found) then write to file
+      fs.writeFile(
+        "/home/jovyan/.config/Code/User/argv.json",
+        '{\n  "locale":"' + lang + '"\n}',
+        function (err) {
+          if (err) throw err;
+          console.log("Updated to current browser language of:" + lang);
+        }
+      );
+    } // end If not found
+
+    res.end(
+      "This port is used to set the language of vsCode being:(" + lang + " )"
+    );
```

Comment:
If you're using async calls, this belongs in a callback function. And then you can send an appropriate HTTP response (e.g. 200 vs. 500) depending on the actual result of the operation.

While sync/async doesn't _really_ matter in this case, the async form would definitely be best practice for an HTTP server.